### PR TITLE
VPC: Extend VPC related API's for v2 VPC reconcile

### PR DESCRIPTION
**Pull request by cjschaef**
Extend the API's used in the v2 support of reconciling VPC for the VPCCluster support.

**What this PR does / why we need it**:  implements the v2 support of reconciling a VPC for VPCCluster support

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
implement new VPC reconcile for VPCClusters
```

### CODE REVIEW

File: api/v1beta2/types.go [hunk 1]
```diff
@@ -400,3 +400,53 @@ type VPCEndpoint struct {
 	// +optional
 	LBID *string `json:"loadBalancerIPID,omitempty"`
 }
+
+// GenericResourceReference represents a generic IBM Cloud resource.
+type GenericResourceReference struct {
```
Comment (mkumatag) [inline api/v1beta2/types.go:405]: I don't see this is used anywhere!

File: api/v1beta2/ibmvpccluster_types.go [hunk 2]
```diff
@@ -163,6 +178,45 @@ type IBMVPCClusterStatus struct {
...
+	// +optional
+	ResourceGroup *ResourceStatus `json:"resourceGroup,omitempty"`
+
+	// vpc references the IBM Cloud VPC.
+	// +optional
+	VPC *VPCResourceStatus `json:"vpc,omitempty"`
+}
+
+// VPCResourceStatus identifies a VPC resource by id (and name) and whether it is ready.
+type VPCResourceStatus struct {
```
Comment (mkumatag) [inline api/v1beta2/ibmvpccluster_types.go:194]: `ResourceStatus` also look the same, can we reuse?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: api/v1beta2/ibmvpccluster_types.go (hunk 2 above)
Comment (cjschaef) [reply api/v1beta2/ibmvpccluster_types.go:194]: I had designed the two status' initially, but can not recall why I had a unique one for VPC resources ATM.
I'll remove and use `ResourceStatus` for all resources instead.

File: api/v1beta2/types.go (hunk 1 above)
Comment (cjschaef) [reply api/v1beta2/types.go:405]: Sorry, too early for that resource, I'll remove it for this PR.

Comment (cjschaef) [reply api/v1beta2/types.go]: Removed

File: api/v1beta2/ibmvpccluster_types.go (hunk 2 above)
Comment (cjschaef) [reply api/v1beta2/ibmvpccluster_types.go]: Removed to use `ResourceStatus`

File: api/v1beta2/ibmvpccluster_types.go [hunk 3]
```diff
@@ -113,7 +113,11 @@ type VPCNetworkSpec struct {
 	// +optional
 	ResourceGroup *string `json:"resourceGroup,omitempty"`
 
-	// TODO(cjschaef): Complete spec definition (SecurityGroups, VPC)
+	// vpc defines the IBM Cloud VPC.
+	// +optional
```
Comment (Karthik-K-N) [inline api/v1beta2/ibmvpccluster_types.go:117]: We already have a field named VPC under IBMVPCClusterSpec, so may be lets add a comment whereever it looks appropriate on whats the difference or when to use which field.

**Review by Karthik-K-N — COMMENTED** [review]
I think its also ok to have one big PR with various small commits, so it gives overall good picture of intent of changes.

File: api/v1beta2/ibmvpccluster_types.go (hunk 3 above)
Comment (cjschaef) [reply api/v1beta2/ibmvpccluster_types.go:117]: I added a little more detail to this field and some other fields/structs, as they related to the extended VPC Infra support.

**Comment by cjschaef** [comment]
Repushing to get CLA to pass.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**